// @angular/forms/src/form_providers.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from '@angular/forms/src/form_providers';
import * as import2 from '@angular/forms/src/directives';
import * as import3 from '@angular/forms/src/directives/radio_control_value_accessor';
import * as import5 from '@angular/forms/src/form_builder';
var FormsModuleInjector = (function (_super) {
    __extends(FormsModuleInjector, _super);
    function FormsModuleInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    Object.defineProperty(FormsModuleInjector.prototype, "_RadioControlRegistry_2", {
        get: function () {
            if ((this.__RadioControlRegistry_2 == null)) {
                (this.__RadioControlRegistry_2 = new import3.RadioControlRegistry());
            }
            return this.__RadioControlRegistry_2;
        },
        enumerable: true,
        configurable: true
    });
    FormsModuleInjector.prototype.createInternal = function () {
        this._InternalFormsSharedModule_0 = new import2.InternalFormsSharedModule();
        this._FormsModule_1 = new import1.FormsModule();
        return this._FormsModule_1;
    };
    FormsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.InternalFormsSharedModule)) {
            return this._InternalFormsSharedModule_0;
        }
        if ((token === import1.FormsModule)) {
            return this._FormsModule_1;
        }
        if ((token === import3.RadioControlRegistry)) {
            return this._RadioControlRegistry_2;
        }
        return notFoundResult;
    };
    FormsModuleInjector.prototype.destroyInternal = function () {
    };
    return FormsModuleInjector;
}(import0.NgModuleInjector));
export var FormsModuleNgFactory = new import0.NgModuleFactory(FormsModuleInjector, import1.FormsModule);
var ReactiveFormsModuleInjector = (function (_super) {
    __extends(ReactiveFormsModuleInjector, _super);
    function ReactiveFormsModuleInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    Object.defineProperty(ReactiveFormsModuleInjector.prototype, "_FormBuilder_2", {
        get: function () {
            if ((this.__FormBuilder_2 == null)) {
                (this.__FormBuilder_2 = new import5.FormBuilder());
            }
            return this.__FormBuilder_2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ReactiveFormsModuleInjector.prototype, "_RadioControlRegistry_3", {
        get: function () {
            if ((this.__RadioControlRegistry_3 == null)) {
                (this.__RadioControlRegistry_3 = new import3.RadioControlRegistry());
            }
            return this.__RadioControlRegistry_3;
        },
        enumerable: true,
        configurable: true
    });
    ReactiveFormsModuleInjector.prototype.createInternal = function () {
        this._InternalFormsSharedModule_0 = new import2.InternalFormsSharedModule();
        this._ReactiveFormsModule_1 = new import1.ReactiveFormsModule();
        return this._ReactiveFormsModule_1;
    };
    ReactiveFormsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.InternalFormsSharedModule)) {
            return this._InternalFormsSharedModule_0;
        }
        if ((token === import1.ReactiveFormsModule)) {
            return this._ReactiveFormsModule_1;
        }
        if ((token === import5.FormBuilder)) {
            return this._FormBuilder_2;
        }
        if ((token === import3.RadioControlRegistry)) {
            return this._RadioControlRegistry_3;
        }
        return notFoundResult;
    };
    ReactiveFormsModuleInjector.prototype.destroyInternal = function () {
    };
    return ReactiveFormsModuleInjector;
}(import0.NgModuleInjector));
export var ReactiveFormsModuleNgFactory = new import0.NgModuleFactory(ReactiveFormsModuleInjector, import1.ReactiveFormsModule);
